packages/block-library/src/post-author/edit.js

```diff
@@ -189,55 +192,65 @@ function PostAuthorEdit( {
 				/>
 			</BlockControls>
 
-			<div { ...blockProps }>
-				{ showAvatar && authorDetails?.avatar_urls && (
-					<div className="wp-block-post-author__avatar">
-						<img
-							width={ attributes.avatarSize }
-							src={
-								authorDetails.avatar_urls[
-									attributes.avatarSize
-								]
-							}
-							alt={ authorDetails.name }
-						/>
-					</div>
-				) }
-				<div className="wp-block-post-author__content">
-					{ ( ! RichText.isEmpty( byline ) || isSelected ) && (
-						<RichText
-							identifier="byline"
-							className="wp-block-post-author__byline"
-							aria-label={ __( 'Post author byline text' ) }
-							placeholder={ __( 'Write byline…' ) }
-							value={ byline }
-							onChange={ ( value ) =>
-								setAttributes( { byline: value } )
-							}
-						/>
+			{ supportsAuthor ? (
+				<div { ...blockProps }>
+					{ showAvatar && authorDetails?.avatar_urls && (
+						<div className="wp-block-post-author__avatar">
+							<img
+								width={ attributes.avatarSize }
+								src={
+									authorDetails.avatar_urls[
+										attributes.avatarSize
+									]
+								}
+								alt={ authorDetails.name }
+							/>
+						</div>
 					) }
-					<p className="wp-block-post-author__name">
-						{ isLink ? (
-							<a
-								href="#post-author-pseudo-link"
-								onClick={ ( event ) => event.preventDefault() }
-							>
-								{ authorName }
-							</a>
-						) : (
-							authorName
+					<div className="wp-block-post-author__content">
+						{ ( ! RichText.isEmpty( byline ) || isSelected ) && (
+							<RichText
+								identifier="byline"
+								className="wp-block-post-author__byline"
+								aria-label={ __( 'Post author byline text' ) }
+								placeholder={ __( 'Write byline…' ) }
+								value={ byline }
+								onChange={ ( value ) =>
+									setAttributes( { byline: value } )
+								}
+							/>
 						) }
-					</p>
-					{ showBio && (
-						<p
-							className="wp-block-post-author__bio"
-							dangerouslySetInnerHTML={ {
-								__html: authorDetails?.description,
-							} }
-						/>
+						<p className="wp-block-post-author__name">
+							{ isLink ? (
+								<a
+									href="#post-author-pseudo-link"
+									onClick={ ( event ) =>
+										event.preventDefault()
+									}
+								>
+									{ authorName }
+								</a>
+							) : (
+								authorName
+							) }
+						</p>
+						{ showBio && (
+							<p
+								className="wp-block-post-author__bio"
+								dangerouslySetInnerHTML={ {
+									__html: authorDetails?.description,
+								} }
+							/>
+						) }
+					</div>
+				</div>
+			) : (
+				<div { ...blockProps }>
+					{ __(
+						'The current post type does not support authors. The Post Author block will not be displayed.'
```

Comment:
Let's use early return for this instead for condition. It should be more readable.

You could do that just below the `showAuthorControl` variable definition.